tests/tiup-cluster/script/scale_core.sh

```diff
@@ -58,10 +58,17 @@ function scale_core() {
     echo "start scale in pd"
     tiup-cluster $client --yes scale-in $name -N $ipprefix.103:2379
     wait_instance_num_reach $name $total_sub_one $native_ssh
+
+    # validate https://github.com/pingcap/tiup/issues/786
+    # ensure that this instance is removed from the startup scripts of other components that need to rely on PD
+    tiup-cluster $client exec $name -N $ipprefix.101 --command "grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh && exit 1 || exit 0"
```

Comment:
This may not work as you expected:

```sh
exit 1 && exit 1 || exit 0 # 1
exit 0 && exit 1 || exit 0 # 0
```
So 
```sh
grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh && exit 1 || exit 0
```
is equivalent with
```
grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh
```
Suggest using:
```
grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh && exit 1
```

Reply:
This is what I'm not expected, and thanks for your kindness advice. 😄

Reply:
Sorry, I'm kind of forgetting where I left off, but it means we need the opposite result(ensure the pd backend was removed from config's file), so we need to rewrite the return code 

```bash
grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh && exit 1 || exit 0"
```

and the below is enough:

```bash
grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh || exit 0"
```

If found, then the return code is the one on left; else the right one(`0`), and that's what we want.

Reply:
```
grep -q $ipprefix.103:2379 /home/tidb/deploy/tidb-4000/scripts/run_tidb.sh || exit 0
```
Will always exit with 0.

If you need the opposite result, I think a single `!` is enough, something like
https://github.com/pingcap/tiup/blob/8aacd48c388f455e80a21bcecca282bdf3e2cc44/tests/tiup-cluster/script/cmd_subtest.sh#L94
(replace the `ls` command with `grep` command)

Reply:
Thanks for the advice.